src/app/patient-dashboard/common/locator-map/locator-map.component.ts

```diff
@@ -92,15 +116,22 @@ export class LocatorMapComponent implements OnInit, OnDestroy {
   public setPhoto() {
     const photo = this.patient.person.getPersonAttribute(this.attributeType);
     if (photo) {
+      console.log(photo);
       this.dataModel = photo;
+      const re = /pdf/gi;
+      if (this.dataModel.search(re) !== -1) {
+        this.setPdfUrl();
+      }
     } else {
       this.dataModel = null;
     }
 
   }
-
-  private getUrl() {
-    return this.appSettingsService.getEtlRestbaseurl().trim() + 'files/';
+  private setPdfUrl() {
+    this.fileUploadResourceService.getFile(this.dataModel, 'pdf').subscribe((file) => {
```

Comment:
we need a 'spinner' or a 'busy indicator' for this async call to the backend, just to show the user that something is happening. Add this to the various calls to the file upload resource service.

Reply:
I have added a  spinner